src/Cli/Microsoft.DotNet.Cli.Utils/MSBuildArgs.cs

```diff
@@ -0,0 +1,155 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Frozen;
+using System.CommandLine;
+
+namespace Microsoft.DotNet.Cli.Utils;
+
+/// <summary>
+/// Represents all of the parsed and forwarded arguments that the SDK should pass to MSBuild.
+/// </summary>
+public class MSBuildArgs
+{
+    private MSBuildArgs(FrozenDictionary<string, string>? properties, FrozenDictionary<string, string>? restoreProperties, string[]? otherMSBuildArgs)
+    {
+        GlobalProperties = properties;
+        RestoreGlobalProperties = restoreProperties;
+        OtherMSBuildArgs = otherMSBuildArgs is not null
+            ? [.. otherMSBuildArgs]
+            : new List<string>();
+    }
+
+    /// <summary>
+    /// The set of `-p` flags that should be passed to MSBuild.
+    /// </summary>
+    public FrozenDictionary<string, string>? GlobalProperties { get; }
```

Comment:
Are FrozenDictionaries the right choice? I think those have quick lookups but slower creation times.

Reply:
I'm OK taking the hit here because it forces correct usage.

Reply:
Wouldn't ImmutableDictionary or ReadOnlyDictionary work as well?

Reply:
You're right - I did some checking and ROD has a better balance of construction speed and lookup speed overall. I'll make that change.